### ISSUE TITLE
Canvas is broken in Safari 8.0.8

### ISSUE THREAD

**Issue by jmoenig** [jmoenig/Snap--Build-Your-Own-Blocks#931]
As I'm researching into #895 I just noticed that the latest version of Safari (8.0.8) has some serious issues in Canvas which I swear weren't there when I last did extensive testing before the Scratch Conference in Amsterdam this year.

Resizing the stage has become excruciatingly slow (and I swear it used to be blazingly fast just a few weeks ago). What's worse, very often the pentrails layer doesn't get rendered at all.

open this sample project in Safari:

http://snap.berkeley.edu/snapsource/snap.html#present:Username=jens&ProjectName=spiral&editMode&noRun

then resize the stage and watch how the pentrails often do not get rendered.

A few years ago Firefox exhibited a similar problem, which I was able to work around by rounding or truncating all coordinates before blitting images (which leads to all kinds of follow-up problems). At least Firefox used to raise some "security" warnings when it didn't draw an image, Safari doesn't seem to even bother.

Looking at the changelog for the latest Safari release it appears they have indeed been tampering with Canvas and some kind of (... obscene language would be appropriate here ...) ```improved canvas taint tracking``` (https://support.apple.com/en-us/HT205033). As Safari doesn't provide any error message at all I can only guess that this might be related to the recent deterioration of Safari's Canvas handling.

Sigh.

**Comment by cycomachead** [comment]
Do you see this still? I think I see it in Safari 11 - though it's not all that often, it only happens at certain stage sizes, so it does seem like it might be a rounding issue.

**Comment by jmoenig** [comment]
Yeah, I'm still seeing this. It seems to have become perhaps slightly better, but overall Safari is still sucking badly. :(
